Source/EasyNetQ/MessageVersioning/MessageVersionStack.cs

```diff
@@ -56,11 +56,18 @@ private static Stack<Type> ExtractMessageVersions( Type type )
 
         private static Type GetSupersededType( Type type )
         {
-             return type
+            var types = FindSupersedes(type);
+            var parentTypes = FindSupersedes(type.BaseType);
+
+            return types.FirstOrDefault(t => !parentTypes.Contains(t));
+        }
+
+        private static IEnumerable<Type> FindSupersedes( Type type )
+        {
+            return type
```

Comment:
There is one corner case: if `MessageVersionStack` will be created from `object` the creator will receive `System.NullReferenceException`.  

Could we check `type` for `null` here?

Reply:
> There is one corner case: if `MessageVersionStack` will be created from `object` the creator will receive `System.NullReferenceException`.
> 
> Could we check `type` for `null` here?

Object is somewhat useless as a message structure, but better safe than sorry. I'll add the check.